regression_em/model.py

```diff
@@ -1,18 +1,287 @@
+       as ``n_samples / (n_classes * np.bincount(y))``
+       Note that these weights will be multiplied with sample_weight (passed
+       through the fit method) if sample_weight is specified.
+
+       feature_index : int
+       the index to devide X into left latent features and right latent features.
+
+       :Example:
+
+       >>> import RegressionEM
+       >>> left_feat = array([[0.17843638, 0.09311004, 0.89600447, ...],
+       [0.55349066, 0.83427622, 0.34841103, ...],
+       ...,
+       [0.22199485, 0.19540406, 0.02678277, ...],
+       [0.62612729, 0.71996384, 0.66445362, ...]])
+       >>> right_feat = array([[0.6177678 , 0.69322733, 0.95146727, ...],
+       [0.96681348, 0.79037145, 0.45834361, ...],
+       ...,
+       [0.64773992, 0.86541352, 0.04755084, ...],
+       [0.37910497, 0.44344932, 0.48168189, ...]])
+       >>> X = np.hstack([left_feat, right_feat])
+       >>> y = array([0, 0, 0, ...,  1, 0, 1])
+       >>> rem = RegressionEM(max_iter=100, with_sample_weights=True, alpha=1)
+       >>> rem.fit(X, y, 100)
+    """
+
+    def __init__(self, alpha: float = 0, max_iter: int = 100, epsilon: float = 10 ** -10, class_weights: str = None, split_index: int = 0) -> None:
+        """Initialize hyper parameters."""
+
+        self._alpha = alpha
+        self.max_iter = max_iter
+        self.epsilon = epsilon
+        self.class_weights = class_weights
+        self.split_index = split_index
+
+    @staticmethod
+    def binary_to_bool(y: np.array) -> np.array:
+        return [True if i > 0 else False for i in y]
+
+    @staticmethod
+    def calc_probs(coef: Matrix, intercept: float, feat_mat: Matrix) -> Matrix:
+        """Return probabilities calculated based on the definition of logistic regression.
+
+        :param coef: Coefficient vector of LR.
+        :param intercept: Intercept parameter of LR.
+        :param feat_vec: Feature vector.
+        :return: Sequence of probabilities.
+        """
+        # 1 / exp(- (feature @ coef + intercept))
+        return 1 / (1 + np.exp(- feat_mat @ coef - intercept))
+
+    def calc_logits(self, probs: Matrix) -> Matrix:
+        """Return logits calculated from probability array.
+
+        :param probs: Probability array to be converted to logists.
+        :param epsilon: Tiny value for clipping.
+        :return: Logist arrray.
+        """
+        # Perform clipping to avoid log(0) and zero division.
+        if sp.issparse(probs):
+            clipped = probs.todense()
+        else:
+            clipped = probs
+        clipped[clipped == 0] = self.epsilon
+        clipped[clipped == 1] = 1 - self.epsilon
+
+        return np.log(clipped / (1 - clipped))
+
+    @staticmethod
+    def calc_responsibility(target_prob: float, ref_prob: float, is_positive: bool) -> float:
+        """Return responsibility to be used in EM algorithm.
+        For detail, see eq.1 of [https://static.googleusercontent.com/media/research.google.com/ja//pubs/archive/46485.pdf].
```

Comment:
Newline here.

Reply:
See [PEP257](https://www.python.org/dev/peps/pep-0257/#id17)

Reply:
Fixed all docstrings to follow PEP257.